src/configuration.ts

```diff
@@ -245,7 +246,21 @@ export class ConnectionConfiguration implements IConnectionConfiguration {
   get pdkGemVerDir(): string {
     var rootDir = path.join(this.pdkRubyDir, 'lib', 'ruby', 'gems');
 
-    return path.join(rootDir, this.gemRubyVersionDir);;
+    return this.findFirstDirectory(rootDir);
+  }
+
+  private findFirstDirectory(rootDir: string) {
+    var files = fs.readdirSync(rootDir);
+    let result = files.sort( (a, b) => a.localeCompare(b, undefined, { numeric:true }) ).reverse()[0];
```

Comment:
This is rather naive, and kinda what got us hardcoding things in the first place. However, the logic works in that the versions are sorted in descending order this time around instead of assuming the one we want is the last in the list. The sorted 'first' is our latest, and the one we should use both for ruby version as well as for the cached version.

This sort can fail if there are weird version numbers that don't match semver, but ruby generally follows these and Puppet doesn't vendor interim versions so PDK doesn't either.

Reply:
I agree this is naive but it may be exactly what we need right now (as opposed to hardcode).  We should definitely prioritise fixing this properly post release.